sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/DocumentHelper.cs

```diff
@@ -0,0 +1,164 @@
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeServerPort)
+                {
+                    serverPort = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeUrlPath)
+                {
+                    urlPath = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeUrlQuery)
+                {
+                    urlQuery = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeHttpResponseStatusCode)
+                {
+                    httpResponseStatusCode = tag.Value.ToString();
+                    continue;
+                }
+            }
+
+            var length = urlScheme.Length + Uri.SchemeDelimiter.Length + serverAddress.Length + serverPort.Length + urlPath.Length + urlQuery.Length;
+            var urlStringBuilder = new System.Text.StringBuilder(length)
+                .Append(urlScheme)
+                .Append(Uri.SchemeDelimiter)
+                .Append(serverAddress)
+                .Append(serverPort != null ? $":{serverPort}" : string.Empty)
+                .Append(urlPath)
+                .Append(urlQuery);
+
+            Request requestDocumentIngress = new()
+            {
+                DocumentType = DocumentIngressDocumentType.Request,
+                Name = activity.DisplayName,
+                Url = urlStringBuilder.ToString(),
+                ResponseCode = httpResponseStatusCode,
+                Duration = activity.Duration < SchemaConstants.RequestData_Duration_LessThanDays
+                                                ? activity.Duration.ToString("c", CultureInfo.InvariantCulture)
+                                                : SchemaConstants.Duration_MaxValue,
+                // TODO: Properties = new Dictionary<string, string>(), - UX supports up to 10 custom properties
+
+                // The following properties are used to calculate metrics. These are not serialized.
+                Extension_IsSuccess = IsSuccess(activity, httpResponseStatusCode),
+                Extension_Duration = activity.Duration.TotalMilliseconds,
+            };
+
+            return requestDocumentIngress;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static Models.Exception CreateException(string exceptionType, string exceptionMessage)
+        {
+            Models.Exception exceptionDocumentIngress = new()
+            {
+                DocumentType = DocumentIngressDocumentType.Exception,
+                ExceptionType = exceptionType,
+                ExceptionMessage = exceptionMessage,
+                // TODO: Properties = new Dictionary<string, string>(), - UX supports up to 10 custom properties
```

Comment:
AI SDK sets first 10 properties from the list.

Reply:
skipping this for now